muspinsim_config/muspinsim_config.xml

```diff
@@ -375,42 +234,80 @@
+            <param name="intrinsic_field" value="0.02"/>
             <param name="polarization_preset" value="custom"/>
             <param name="polarization" value="1 0 0"/>
             <param name="time" value="range(0,8.0,1000)"/>
             <param name="time" value="range(0,1.0)"/>
             <param name="orientation_preset" value="eulrange"/>
             <param name="eul_n" value="10"/>
-            <output name="out_file" file="test_4.in" ftype="txt" compare="diff" />
+            <param name="celio_enabled" value="true"/>
+            <param name="celio_k" value="10"/>
+            <output name="out_file" file="test_4.in" ftype="txt" compare="diff"/>
         </test>
         <test expect_num_outputs="1">
             <param name="out_file_prefix" value="test_5"/>
             <param name="spin_preset" value="mu"/>
             <param name="interaction" value="dissipation"/>
             <param name="dis_index" value="1"/>
             <param name="dis_val" value="g"/>
-            <param name="experiment_preset" value="custom" />
-            <param name="field" value="1.0/muon_gyr" />
+            <param name="experiment_preset" value="custom"/>
+            <param name="field" value="1.0/muon_gyr"/>
+            <param name="intrinsic_field" value="2.0/muon_gyr"/>
             <param name="fitting" value="true"/>
-            <param name="var_name" value="g" />
-            <param name="min_bound" value="0.0" />
-            <param name="fitting_tolerance" value="1.0" />
-            <output name="out_file" file="test_5.in" ftype="txt" compare="diff" />
+            <param name="var_name" value="g"/>
+            <param name="min_bound" value="0.0"/>
+            <param name="fitting_tolerance" value="1.0"/>
+            <param name="celio_enabled" value="true"/>
+            <param name="celio_k" value="10"/>
+            <param name="celio_averages" value="4"/>
+            <output name="out_file" file="test_5.in" ftype="txt" compare="diff"/>
+        </test>
+        <test expect_num_outputs="1">
+            <param name="out_file_prefix" value="test_6"/>
+            <param name="experiment_preset" value="zero_field"/>
+            <param name="spin_preset" value="custom"/>
+            <param name="spin" value="H"/>
+            <param name="spin_preset" value="mu"/>
+            <param name="interaction" value="zeeman"/>
+            <param name="zeeman_index" value="1"/>
+            <param name="zeeman_vector" value="1 0 0"/>
+            <param name="zeeman_index" value="2"/>
+            <param name="zeeman_vector" value="2 0 0"/>
+            <param name="average_axes" value="orientation,temperature"/>
+            <output name="out_file" file="test_6.in" ftype="txt" compare="diff"/>
+        </test>
+        <test expect_num_outputs="1">
+            <param name="out_file_prefix" value="test_6"/>
+            <param name="experiment_preset" value="zero_field"/>
```

Comment:
```suggestion
            <param name="experiment_preset" value="custom"/>
```
Can we make this a test of `custom`? The bug I observed was only with that setup, currently this test wouldn't reproduce the bug since it's using the `zero_field` logic like 6.